src/Education/Commands/RuleDescriptionWindowCommand.cs

```diff
@@ -0,0 +1,82 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Design;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace SonarLint.VisualStudio.Education.Commands
+{
+    internal sealed class RuleDescriptionWindowCommand
+    {
+        public static RuleDescriptionWindowCommand Instance { get; private set; }
+
+        private const int commandId = 0x100;
+        private readonly IToolWindowService toolWindowService;
+        private readonly ILogger logger;
+
+        internal static readonly Guid commandSet = new Guid("80127033-1819-4996-8C45-E9C96F75E2A8");
+
+        internal RuleDescriptionWindowCommand(IMenuCommandService commandService, IToolWindowService toolWindowService, ILogger logger)
+        {
+            commandService = commandService ?? throw new ArgumentNullException(nameof(commandService));
+
+            this.toolWindowService = toolWindowService ?? throw new ArgumentNullException(nameof(toolWindowService));
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger)); ;
+
+            var menuCommandID = new CommandID(commandSet, commandId);
+            var menuItem = new MenuCommand(this.Execute, menuCommandID);
+
+            commandService.AddCommand(menuItem);
+        }
+
+        public static async Task InitializeAsync(AsyncPackage package)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(package.DisposalToken);
+
+            var commandService = await package.GetServiceAsync(typeof(IMenuCommandService)) as IMenuCommandService;
+            var componentModel = await package.GetServiceAsync(typeof(SComponentModel)) as IComponentModel;
+
+            var toolWindowService = componentModel.GetService<IToolWindowService>();
+            var logger = componentModel.GetService<ILogger>();
+
+            Instance = new RuleDescriptionWindowCommand(commandService, toolWindowService, logger);
+        }
+
+        internal void Execute(object sender, EventArgs e)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            try
+            {
+                toolWindowService.Show(RuleDescriptionToolWindow.ToolWindowId);
+            }
+            catch (Exception ex) when (!Core.ErrorHandler.IsCriticalException(ex))
+            {
+                logger.WriteLine(ex.ToString());
```

Comment:
IssueVisCommand has this instead of just toString, however I do not know if the message in Resources is generic or not. Please let me know so I can fix this

logger.WriteLine(string.Format(Resources.ERR_VisualizationToolWindow_Exception, ex));